test/e2e/specs/convert-block-type.test.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Internal dependencies
+ */
+import {
+	getEditedPostContent,
+	newPost,
+	insertBlock,
+} from '../support/utils';
+
+describe( 'Code block', () => {
+	beforeEach( async () => {
+		await newPost();
+	} );
+
+	it( 'should convert to a preformatted block', async () => {
+		const code = 'print "Hello Dolly!"';
+
+		await insertBlock( 'Code' );
+
+		await page.type( '.editor-block-list__block textarea', code );
+
+		// Verify the content starts as a Code block.
+		const originalPostContent = await getEditedPostContent();
+		const codeRegex = new RegExp( `<!-- wp:code -->\\s*<pre class="wp-block-code"><code>${ code }<\\/code><\\/pre>\\s*<!-- \\/wp:code -->` );
```

Comment:
I think we should probably have a helper for this pattern, I noticed it used elsewhere and it's a bit of a pain. I can refactor this and convert it here if you like.

Reply:
Can we do the same check using a simpler CSS selector?

Reply:
We want to check for the entire content, including the comment, so I don't think so.

Reply:
Why is this not a snapshot?

Reply:
Oops, it totally should be. Good point, I’ll change it in a follow-up patch.

I’m just not used to working with snapshots in Jest so I don’t think to use them. Sorry!

Reply:
Addressed in https://github.com/WordPress/gutenberg/pull/10166